python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -980,42 +980,24 @@ def resolve_latestMaterializationByPartition(
             lambda event: event.dagster_event.step_materialization_data.materialization.partition
         )
 
-        query_all_partitions = partitions is None
-        partitions = self.get_partition_keys() if query_all_partitions else partitions
-
-        if query_all_partitions or len(partitions) > 1000:
-            # when there are many partitions, it's much more efficient to grab the latest storage
-            # id for all partitions and then query for the materialization events based on those
-            # storage ids
-            latest_storage_ids = sorted(
-                (
-                    graphene_info.context.instance.event_log_storage.get_latest_storage_id_by_partition(
-                        self._asset_node_snap.asset_key, DagsterEventType.ASSET_MATERIALIZATION
-                    )
-                ).values()
-            )
-            events_for_partitions = get_asset_materializations(
-                graphene_info,
-                asset_key=self._asset_node_snap.asset_key,
-                storage_ids=latest_storage_ids,
-            )
-            latest_materialization_by_partition = {
-                get_partition(event): event for event in events_for_partitions
-            }
```

Comment:
the change being made in this PR seems like a big improvement for the case where you pass in 1001 partitions and there are 10000 partitions - we were fetching a bunch of materializations before in that case that we then just threw out and didn't use right?